lib/SILOptimizer/IPO/CrossModuleOptimization.cpp

```diff
@@ -138,10 +139,10 @@ class InstructionVisitor : public SILCloner<InstructionVisitor> {
 
 private:
   CrossModuleOptimization &CMS;
-
+  bool checkSerializableInst;
 public:
-  InstructionVisitor(SILFunction &F, CrossModuleOptimization &CMS) :
-    SILCloner(F), CMS(CMS) {}
+  InstructionVisitor(SILFunction &F, CrossModuleOptimization &CMS, bool checkSerializableInst) :
```

Comment:
IMO this is a bad name because it suggests that the check is done _in addition_ to the other work. How about `onlyCheckIfSerializable`?

Reply:
updated to use VisitorMode enum.